backend/session/logind.c

```diff
@@ -104,15 +104,17 @@ static void logind_release_device(struct wlr_session *base, int fd) {
 	struct logind_session *session = logind_session_from_session(base);
 
 	struct stat st;
-	if (fstat(fd, &st) < 0) {
+	int ret = fstat(fd, &st);
+	close(fd);
+	if (ret < 0) {
 		wlr_log(WLR_ERROR, "Failed to stat device '%d': %s", fd,
 			strerror(errno));
```

Comment:
Can we move the `strerror` call before `close`, to prevent it from showing an erroneous value?

Reply:
Good point. Wetson has the same bug: https://gitlab.freedesktop.org/wayland/weston/-/blob/master/libweston/launcher-logind.c#L229.